src/main/java/players/rhea/RHEAIndividual.java

```diff
@@ -0,0 +1,177 @@
+     */
+    public int rollout(AbstractGameState gs, AbstractForwardModel fm, int startIndex, int endIndex, int playerID) {
+        length = 0;
+        int fmCalls = 0;
+        double delta = 0;
+        for (int i = 0; i < startIndex; i++) {
+            double score;
+            if (this.heuristic != null){
+                score = heuristic.evaluateState(gameStates[i+1], playerID);
+            } else {
+                score = gameStates[i+1].getHeuristicScore(playerID);
+            }
+
+            delta += Math.pow(discountFactor, i) * score;
+        }
+        for (int i = startIndex; i < endIndex; i++){
+            // Rolls from chosen index to the end, randomly changing actions and game states
+            // Length of individual is updated depending on if it reaches a terminal game state
+            if (gs.isNotTerminal()) {
+                // Copy the game state
+                AbstractGameState gsCopy = gs.copy();
+                List<AbstractAction> currentActions = fm.computeAvailableActions(gsCopy);
+                AbstractAction action = null;
+                if (currentActions.size() > 0) {
+                    action = currentActions.get(gen.nextInt(currentActions.size()));
+                }
+
+                // Advance game state with random action
+                fm.next(gsCopy, action);
+                fmCalls ++;
+
+                // If it's my turn, store this in the individual
+                boolean iAmMoving = (gameStates[i].getCurrentPlayer() == playerID);
+                if (iAmMoving) {
+                    gameStates[i + 1] = gsCopy;
+                    actions[i] = action;
+
+                    // Individual length increased
+                    length++;
+
+                    // Add value of state, discounted
+                    double score;
+                    if (this.heuristic != null){
+                        score = heuristic.evaluateState(gameStates[i+1], playerID);
+                    } else {
+                        score = gameStates[i+1].getHeuristicScore(playerID);
+                    }
+                    delta += Math.pow(discountFactor, i) * score;
+                } else {
+                    i--;
+                }
+
+                gs = gsCopy;
+            } else {
+                break;
+            }
+        }
+//        this.value = gs.getScore(playerID);
+        this.value = delta;
+        return fmCalls;
+    }
+
+    @Override
+    public int compareTo(Object o) {
+        RHEAIndividual a = this;
+        RHEAIndividual b = (RHEAIndividual)o;
+        return Double.compare(b.value, a.value);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof Individual)) return false;
+
+        RHEAIndividual a = this;
+        RHEAIndividual b = (RHEAIndividual)o;
+
+        for (int i = 0; i < actions.length; i++) {
+            if (!a.actions[i].equals(b.actions[i])) return false;
+        }
```

Comment:
Hence if two individuals have the same sets of actions, but these led to different states (in a stochastic game, or if we redeterminesed for each), then they will be treated as identical.
Probably correct - checking this is what you intended?